### PR TITLE
Add Javadoc comments

### PR DESCRIPTION
**Pull request by Andre601**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [ ] Library interface (affecting end-user code) 
- [x] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description
Pull Request #1178 was labelled with "documentation needed" which is why I make this PR.
So far I only added Javadoc comments to the `JDABuilder#setGatewayEnabled(boolean)` method and didn't touch the RestAction part yet due to not knowing if they need docs and what I should write there (This is also the reason why this is a draft PR)

### CODE REVIEW

**Comment by Andre601** [comment]
> We are going to need more expansive information about what this is actually doing as the documentation here is kinda limited.

> So far I only added Javadoc comments to the JDABuilder#setGatewayEnabled(boolean) method and didn't touch the RestAction part yet due to not knowing if they need docs and what I should write there (This is also the reason why this is a draft PR)

**Comment by MinnDevelopment** [comment]
The average library end user will not understand what "gateway" means. The description has to be a bit more explicit.

File: src/main/java/net/dv8tion/jda/api/JDABuilder.java [hunk 1]
```diff
@@ -138,6 +138,18 @@ public JDABuilder(@Nonnull AccountType accountType)
         this.listeners = new LinkedList<>();
     }
 
+    /**
+     * Whether JDA should connect to the gateway. Setting this to {@code false} will set the {@link net.dv8tion.jda.api.JDA.Status JDA Status} to {@code REST_ONLY}.
+     * <br>In {@code REST_ONLY} mode will JDA not connect to the gateway of Discord and only rely on rest actions performed from/to Discord.
```
Comment (TheDGOfficial) [inline src/main/java/net/dv8tion/jda/api/JDABuilder.java]:
```suggestion
     * <br>In {@code REST_ONLY} mode JDA will not connect to the gateway of Discord and only rely on rest actions performed from/to Discord.
```

**Comment by Andre601** [comment]
I honestly close this PR, as there most likely other people who can draft better Javadoc comments.

**Pull request closed without merging**